src/pages/EarthEngineImport/EarthEngineImportForm.js

```diff
@@ -0,0 +1,337 @@
+                aggregationType,
+            })
+
+            setEeData(structuredData)
+            setPointOuRows(pointOrgUnits)
+            setFetching(false)
+            setDoSubmit(true)
+        } catch (error) {
+            console.log('Error while fetching Earth Engine data', error)
+            setEeData([])
+            setFetching(false)
+            const message = error.message || error
+
+            let msg = i18n.t(
+                'An error occurred while trying to fetch Earth Engine data'
+            )
+
+            if (message.includes('Output of image computation is too large')) {
+                msg = i18n.t(
+                    'The Earth Engine data set is too large. Try reducing the number of groups or organisation units'
+                )
+            } else if (
+                message.includes(
+                    'Dimension is present in query without any valid dimension options: `ou`'
+                )
+            ) {
+                msg = i18n.t('The organisation units selection is invalid')
+            } else if (message.length) {
+                msg = msg.concat(`: ${message}`)
+            }
+            setRequestFailedMessage(msg)
+
+            return
+        }
+    }
+
+    const previewIsAllowed = ({ valid, values, modifiedSinceLastSubmit }) => {
+        // there should be at least one band for Population Age groups
+        const bandsValid =
+            values.earthEngineId === POPULATION_AGE_GROUPS_DATASET_ID
+                ? !!values.bandCocs?.find((bc) => !!bc.coc)
+                : true
+
+        const otherCheck = requestFailedMessage ? modifiedSinceLastSubmit : true
+
+        return valid && bandsValid && otherCheck
+    }
+
+    const wrappedSetShowFullSummaryTask = (val) => {
+        setShowFullSummaryTask(val)
+        hiddenTopElRef?.current?.scrollIntoView({ behavior: 'smooth' })
+    }
+
+    const onImportInternal = onImport({
+        engine,
+        setProgress,
+        addTask,
+        setShowFullSummaryTask: wrappedSetShowFullSummaryTask,
+    })
+
+    const onSubmit = (values) => {
+        if (doSubmit) {
+            onImportInternal({ eeData, ...values })
+        }
+    }
+
+    const getAlerts = (submitError) => {
+        const internalErrors = []
+        if (requestFailedMessage) {
+            internalErrors.push({
+                id: 'getEeDataFailed',
+                message: requestFailedMessage,
+                critical: true,
+                info: false,
+                warning: false,
+            })
+        }
+        return Array.isArray(submitError)
+            ? submitError.concat(internalErrors)
+            : internalErrors
+    }
+
+    return (
+        <>
+            <div ref={hiddenTopElRef} className={styles.hiddenTopElement}></div>
+            <Page
+                title={i18n.t('Earth Engine import')}
+                desc={i18n.t(
+                    'Import Earth Engine data to data sets and data elements'
```

Comment:
Suggested change: "Import WorldPop population data from Google Earth Engine to a DHIS2 data element."